src/components/Technologies/Technologies.tsx

```diff
@@ -2,79 +2,152 @@ import React, { useState, useMemo } from "react";
-import CategorySelector from "./CategorySelector";
-import { PacksData, IntegrationsData } from "../Integrations/IntegrationTypes";
+import { PacksData, IntegrationsData, FrontMatterData } from "../Integrations/IntegrationTypes";
 import TechnologyCard from "./TechnologyCard";
+import PacksFilters from "./PacksFilters";
+import { packTypeNames } from "./PackConstants";
+import { Collapse, ConfigProvider, ThemeConfig, theme } from "antd";
+import "./technologies.antd.css";
+import IconMapper from "../IconMapper/IconMapper";
+import { useColorMode } from "@docusaurus/theme-common";
 
 const searchOptions = {
   threshold: 0.5,
-  keys: ["fields.title"],
+  keys: ["title"],
 };
 
 interface TechnologiesProps {
   data: PacksData[] | IntegrationsData[];
+  repositories: any[];
 }
 
-export default function Technologies({ data }: TechnologiesProps) {
-  const [selectedCategory, setSelectedCategory] = useState("all");
+export default function Technologies({ data, repositories }: TechnologiesProps) {
+  const { isDarkTheme } = useColorMode();
+  const { defaultAlgorithm, darkAlgorithm } = theme;
+  const [selectedFilters, setSelectedFilters] = useState<{ category: string[], registries: string[], provider: string, additionalFilters: string[] }>({ category: [], registries: [], provider: "", additionalFilters: [] })
   const [searchValue, setSearchValue] = useState("");
 
   const categories = useMemo(() => {
-    const categoriesSet = data.reduce(
-      (accumulator, technology) => {
-        const categories = technology.fields.category || [];
-        categories.forEach((category) => {
-          accumulator.add(category);
-        });
-        return accumulator;
-      },
-      new Set(["all"])
-    );
-    return new Set([...categoriesSet].sort());
+    const categoriesMap = data.reduce((acc: Map<string, any>, technology: PacksData | IntegrationsData) => {
+      let packType = technology.fields.packType;
+      if (acc.has(packType)) {
+        acc.get(packType).push(technology.fields);
+      } else {
+        acc.set(packType, [technology.fields]);
+      }
+      return acc;
+    }, new Map<string, any>());
+    const sortedCategoriesMap = new Map([...categoriesMap.entries()].sort());
+    const categoryKeys = Array.from(sortedCategoriesMap.keys());
+    categoryKeys.forEach((category) => {
+      const fields = sortedCategoriesMap.get(category);
+      fields.sort((field1: any, field2: any) => {
+        if (field1.name > field2.name) {
+          return 1;
+        } else if (field1.name < field2.name) {
+          return -1;
+        } else {
+          return 0;
+        }
+      });
+    })
+    return sortedCategoriesMap;
   }, [data]);
 
-  const technologies = useMemo(() => {
-    let technologies = [...data].sort((pack1, pack2) => {
-      const category1 = pack1.fields.category[0];
-      const category2 = pack2.fields.category[0];
-
-      if (category1 < category2) {
-        return -1;
+  const filteredTechCards = useMemo(() => {
```

Comment:
this entire block is very hard to read.
My suggestion is to create a list of conditions that each card would have to accommodate.

```
const conditions = [];

// ....
if (selectedFilters.provider) {
   conditions.push(card => card.cloudTypes.includes(selectedFilters.provider))
}
// ...

// apply the filter only once
cards.filter(card => {
  return conditions.every(condition => condition(card))
})

Reply:
at the end you can use the fuse search to narrow things down even further and then group them by categories

Reply:
I'm not sure if your approach here to have the data already grouped by categories is the best.
Flat structures are easier to manipulate.